cmd/gateway-sia-cachelayer.go

```diff
@@ -0,0 +1,1194 @@
+package cmd
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/NebulousLabs/Sia/api"
+	_ "github.com/mattn/go-sqlite3" // Has to be imported as blank
```

Comment:
Lets actually explore on removing this dependency and moving to either https://github.com/pingcap/tidb or https://github.com/boltdb/bolt 

BoltDB is more native in its feel APIs are sort of synonymous and its quite small. 

https://github.com/pingcap/tidb is more feature rich and SQL support. 

I am leaning towards BoltDB but its your choice. We really need to get rid of CGO dependency.